srcCxx/rti_connext_dds/ShapeType_configurator_rti_connext_dds.h

```diff
@@ -49,68 +49,142 @@ class ShapeTypeConfigurator {
             bool use_security,
             const char *governance_file,
             const char *permissions_file,
+            const char *key_establishment_algorithm,
             bool enable_logging)
     {
         DomainParticipant *participant = NULL;
 
         DDS_DomainParticipantQos pQos;
-        TheParticipantFactory->get_participant_qos_from_profile(pQos, "BuiltinQosLib", "Generic.Security");
-        //get_default_participant_qos(pQos);
+        TheParticipantFactory->get_participant_qos_from_profile(
+                pQos,
+                "BuiltinQosLib",
+                "Generic.Security");
 
-         if (!use_security)
-           printf("TODO: Handle !use_security...\n");
+        if (!use_security) {
+            fprintf(stderr, "TODO: Handle !use_security...\n");
+        }
 
         pQos.transport_builtin.mask = DDS_TRANSPORTBUILTIN_UDPv4;
 
-        DDS_PropertyQosPolicyHelper_assert_pointer_property(&pQos.property,
-                 RTI_SECURITY_BUILTIN_PLUGIN_PROPERTY_NAME ".create_function_ptr",
-                 (void *) RTI_Security_PluginSuite_create);
+        if (DDS_PropertyQosPolicyHelper_assert_pointer_property(
+                &pQos.property,
+                RTI_SECURITY_BUILTIN_PLUGIN_PROPERTY_NAME
+                        ".create_function_ptr",
+                (void *) RTI_Security_PluginSuite_create) != DDS_RETCODE_OK) {
+            fprintf(stderr, "Error asserting .create_function_ptr property.\n");
+            return NULL;
```

Comment:
I think we're leaking `pQos` in general.
```suggestion
            goto done;
```
and finalize `pQos` after `done:`.

Reply:
**Note to self**: This is  the C `DDS_DomainParticipantQos` structure and the traditional C++ APIs that are called. It is not the `DomainParticipantQos` modern C++ object which should be destructed when going out of scope.